source/usage/other/craftscripts.rst

```diff
@@ -47,54 +45,128 @@ Scripts have the following three variables in their global namespace:
 
-The ``CraftScriptContext`` can to some basic argument parsing with ``CraftScriptContext#getBlock()``. You can also hook directly into WorldEdit's parsers via ``WorldEdit.getInstance().getPatternFactory()`` and ``.getMaskFactory()``.
+The ``CraftScriptContext`` can to some basic argument parsing with ``CraftScriptContext#getBlock``. You can also hook directly into WorldEdit's parsers via ``WorldEdit.getInstance().getPatternFactory()`` and ``.getMaskFactory()``.
 
 .. topic:: Example: Checking arguments
 
     ::
 
+        // This script accepts at least 1 and at most 3 required arguments, separated by a space.
+        // Use -1 as the second argument for no maximum limit.
         context.checkArgs(1, 3, "<block> [width] [height]");
         var block = context.getBlock(argv[1]);
+        var width = 5;
+        var height = 5;
+        if (argv.length >= 3) {
+          width = parseInt(argv[2])
+        }
+        if (argv.length >= 4) {
+          height = parseInt(argv[3])
+        }
+
+If the player inputs an invalid block, then an exception will be raised. If the exception is not caught, then the player will be informed about the error, and your script will be halted.
+
+Printing Output
+--------------------
+
+Sometimes, you may want to write output to chat. Perhaps you would like to notify that the script has completed running, or perhaps you are debugging your script. In any case, you can do so through ``CraftScriptContext#print``:
 
-What happens if the user inputs an invalid block? An exception will be raised and if you don't catch it, the user will be informed about their error and your script will be halted.
+::
+
+    context.print("Hello!")
 
-Working with Java packages
---------------------------
+Other printing functions, such as ``CraftScriptContext#error`` and ``CraftScriptContext#printRaw``, also exist. This code snippet will print the exception raised if the player inputs an invalid block:
 
-To import a java package, you can use the following syntax::
+.. topic:: Example: Printing an exception to error output
 
-    importPackage(Packages.package.name.here);
+    ::
+
+        context.checkArgs(1, 1, "<block>");
+        try {
+          var block = context.getBlock(argv[1]);
+        } catch (e) {
+          context.error(e);
+        }
+
+Note that ``console.log`` will not work because there is no console, and ``System.out.println`` will not work because the ``System`` class is in the ``java.lang.*`` package and thus cannot be imported.
+
+
+Miscellaneous
+---------------
 
-You can import any package available in the Java classpath - not restricted to WorldEdit.
+Rhino engine doesn't support the ``let`` keyword yet. This means that every variable you declare is global.
```

Comment:
`let` has nothing to do with whether a variable is global or not -- it's strictly about variable hoisting mechanics in JS. I think this statement should simply be:
```suggestion
Rhino does not support all ES2015+ features yet, see https://mozilla.github.io/rhino/compat/engines.html for details.
```

Also, as you can see by looking at the table, modern versions of Rhino actually _do_ support `let`!